extender.go

```diff
@@ -298,16 +298,22 @@ func (e *Extender) extend(kind string, baseImage v1.Image, logger log.Logger) (v
 		return nil, fmt.Errorf("getting image config: %w", err)
 	}
 	buildOptions := e.extendOptions()
+	var userID, origUserID, prevUserID, groupID string
+	userID, groupID = userFrom(*configFile)
+	prevUserID = userID
 	for _, dockerfile := range dockerfiles {
-		userID, groupID := userFrom(*configFile)
 		dockerfile.Args = append([]extend.Arg{
 			{Name: argBuildID, Value: uuid.New().String()},
 			{Name: argUserID, Value: userID},
 			{Name: argGroupID, Value: groupID},
 		}, dockerfile.Args...)
-
+		if origUserID == "" {
+			origUserID = userID
+		}
 		// apply Dockerfile
-
+		if userID == "0" {
+			logger.Warnf("Extension from %s changed the user ID from %d to %d; this must not be the final user ID (a following extension must reset the user).", prevUserID, userID, dockerfile.Path)
+		}
```

Comment:
This should go after `userID, groupID = userFrom(*configFile)` ...I think

Reply:
hmm - if you move it below that line in the loop then you miss the first userID entirely . but you're right that if it's here in the loop then you miss the last user ID - but that's the one that we'll raise an error on. 

So options include:
- (1) keep it here / where it is
pros:  you only have this check once, and every case is covered by _something_
cons: it's a little unintuitive, and you won't get a warning before your error

- (2) move a copy of this above the loop to cover the initial case, and a copy of this down in the loop to cover the updates.
pros: you always get the same warning, including before the error occurs
cons: you have slightly duplicated logic. 


IIRC this whole situation is why languages like C have "do...while" loops so that you can have your initialization happen inside the loop,  but 🤷 

I think I'll try option 2 above since it's the most consistent UX.

Reply:
oh... i see. it doesn't apply on the initial case, since the initial user ID doesn't come from an extension so it can't be root!